### PR TITLE
Fix Migration Error Codes

### PR DESCRIPTION
**Pull request by daryakaviani**
# Task: Fix Migration Error Codes

## Description

- Crab migration error codes were mixed up due to merges
- Fixing tests and contract to use correct error codes

### CODE REVIEW

**Review by aleone — APPROVED** [review]
lgtm

File: packages/hardhat/contracts/strategy/CrabMigration.sol [hunk 1]
```diff
@@ -19,18 +19,17 @@ import {StrategyMath} from "./base/StrategyMath.sol";
...
- * M2: Migration already happened
- * M3: Migration has not yet happened
- * M4: msg.sender is not Euler Mainnet Contract
- * M5: msg. sender cannot send ETH
- * M6: Can't withdraw more than you own
- * M7: Not enough ETH to repay the loan
+ * M1: Migration already happened
+ * M2: Migration has not yet happened
+ * M3: msg.sender is not Euler Mainnet Contract
+ * M4: msg.sender cannot send ETH
```
Comment (alpinechicken) [inline packages/hardhat/contracts/strategy/CrabMigration.sol:25]: This is for the receive right?
`    receive() external payable {
        require(msg.sender == weth || msg.sender == crabV1 || msg.sender == crabV2, "M4");
    }`

Comment (aleone) [reply packages/hardhat/contracts/strategy/CrabMigration.sol:25]: Yeah it is. I think the meaning is that msg.sender (contract that is sending ETH) isnt allowed to, but maybe could be clearer.

File: packages/hardhat/contracts/strategy/CrabMigration.sol [hunk 2]
```diff
@@ -19,18 +19,17 @@ import {StrategyMath} from "./base/StrategyMath.sol";
...
+ * M5: Can't withdraw more than you own
+ * M6: Not enough ETH to repay the loan
+ * M7: Invalid crabV2 address
```
Comment (alpinechicken) [inline packages/hardhat/contracts/strategy/CrabMigration.sol:28]: Mixed with M8? 
        require(_ethToBorrow > 0 && _withdrawMaxEthToPay > 0, "M7");

M8: _ethToBorrow or _withdrawMaxEthToPay can't be 0

Comment (daryakaviani) [reply packages/hardhat/contracts/strategy/CrabMigration.sol:28]: I believe the require statement is `require(_ethToBorrow > 0 && _withdrawMaxEthToPay > 0, "M8")` and tests also use M8?

File: packages/hardhat/contracts/strategy/CrabMigration.sol (hunk 1 above)
Comment (daryakaviani) [reply packages/hardhat/contracts/strategy/CrabMigration.sol]: Agreed; now updated to "Can only receive ETH from weth, crabv1, or crabv2 contract".

File: packages/hardhat/contracts/strategy/CrabMigration.sol (hunk 2 above)
Comment (alpinechicken) [reply packages/hardhat/contracts/strategy/CrabMigration.sol:28]: ah you're right - was looking at main

Comment (aleone) [reply packages/hardhat/contracts/strategy/CrabMigration.sol:28]: What line do you see ```require(_ethToBorrow > 0 && _withdrawMaxEthToPay > 0, "M7");```

I see that being M8 in the code and in the error legend.

**Review by alpinechicken — APPROVED** [review]
lgtm

**Pull request merged**